### PR TITLE
fix logic thread behavior

### PR DESCRIPTION
**Pull request by majestrate**
the logic thread seemed to work in a way that was different from the way i remember it should.

fixed.

### CODE REVIEW

File: llarp/ev/ev.hpp [hunk 1]
```diff
@@ -671,7 +671,8 @@ namespace llarp
 #ifdef _WIN32
 struct llarp_fd_promise
 {
-  void Set(std::pair< int, int >)
+  void
```
Comment (tewinget) [inline llarp/ev/ev.hpp]: is `make format` capricious, or did someone forget to run it at some point?

Comment (majestrate) [reply llarp/ev/ev.hpp]: `make format` should use `clang-format-8` as 9 and higher makes the source get fugged some how.

File: llarp/util/thread/threadpool.cpp [hunk 2]
```diff
@@ -41,64 +36,29 @@ llarp_threadpool_stop(struct llarp_threadpool *pool)
...
+bool
 llarp_threadpool_queue_job(struct llarp_threadpool *pool,
-                           std::function< void() > func)
+                           std::function< void(void) > func)
 {
-  if(pool->impl)
-  {
-    while(!pool->impl->tryAddJob(func))
-    {
-      std::this_thread::sleep_for(std::chrono::microseconds(1000));
```
Comment (notlesh) [inline llarp/util/thread/threadpool.cpp]: This was likely a big source of latency...

File: llarp/util/thread/threadpool.h [hunk 3]
```diff
@@ -5,48 +5,51 @@
...
+
+  size_t
+  pendingJobs() const;
+
+  size_t
+  numThreads() const;
+
+  /// try to guess how big our job latency is on this threadpool
+  llarp_time_t
+  GuessJobLatency(llarp_time_t granulairty = 1000) const;
```
Comment (notlesh) [inline llarp/util/thread/threadpool.h]: A high level description of granularity would be useful. Also, typo (granulairty)

File: llarp/util/thread/threadpool.h [hunk 4]
```diff
@@ -5,48 +5,51 @@
...
+
+  /// see if this thread is full given lookahead amount
+  bool
```
Comment (notlesh) [inline llarp/util/thread/threadpool.h]: Might be better named "HasSpareCapacity()" or similar

File: test/link/test_llarp_link.cpp [hunk 5]
```diff
@@ -133,6 +142,9 @@ struct LinkLayerTest : public test::LlarpTest< llarp::sodium::CryptoLibSodium >
   void
   SetUp()
   {
+    oldLevel = llarp::LogContext::Instance().minLevel;
+    if(shouldDebug)
+      llarp::SetLogLevel(eLogDebug);
```
Comment (notlesh) [inline test/link/test_llarp_link.cpp]: lol...

**Pull request merged**